granary/reddit.py

```diff
@@ -0,0 +1,218 @@
+# coding=utf-8
+"""reddit.com source class.
+
+reddit API docs:
+https://github.com/reddit-archive/reddit/wiki/API
+https://www.reddit.com/dev/apis
+praw API docs:
+https://praw.readthedocs.io/en/v3.6.0/pages/oauth.html
+"""
+
+from . import source
+import logging
+from oauth_dropins import reddit
+from oauth_dropins.webutil import appengine_info, util
+import re
+import urllib.parse, urllib.request
+
+import praw
+
+if appengine_info.DEBUG:
+  REDDIT_APP_KEY = util.read('reddit_app_key_local')
+  REDDIT_APP_SECRET = util.read('reddit_app_secret_local')
+else:
+  REDDIT_APP_KEY = util.read('reddit_app_key')
+  REDDIT_APP_SECRET = util.read('reddit_app_secret')
+
+
+def get_reddit_api(refresh_token):
+  return praw.Reddit(client_id=REDDIT_APP_KEY,
+             client_secret=REDDIT_APP_SECRET,
+             refresh_token=refresh_token,
+             user_agent='oauth-dropin reddit api')
+
+
+class Reddit(source.Source):
+
+  DOMAIN = 'reddit.com'
+  BASE_URL = 'https://reddit.com'
+  NAME = 'reddit'
```

Comment:
very minor, mind capitalizing? looks like they do themselves on their home page. and probably `RedditAuth.site_name()` in o-d too.